lib/MetaCPAN/Document/File.pm

```diff
@@ -705,6 +705,9 @@ sub set_indexed {
     my ( $self, $meta ) = @_;
 
     foreach my $mod ( @{ $self->module } ) {
+        if ( $mod->name =~ /^[_*]/ ) {
```

Comment:
It isn't obvious why the asterisk is here... it makes me wonder if that was intentional.
Instead of special casing some exclusions (leading underscore), perhaps it would be better to use the [same regexp that PAUSE does](https://github.com/andk/pause/blob/master/lib/PAUSE/pmfile.pm#L249) and simply invert the check:
`if( $mod->name !~ /^[A-Za-z]/ ){ $mod->indexed(0) }`

I would additionally then comment on why we're doing this check (where the regexp came from (link to the PAUSE source code)).

The test should still pass, but the code will be clearer and more representative of where the "rule" came from.